fs/metrics/common/metrics.go

```diff
@@ -62,22 +62,61 @@ const (
 
 // Lists all metric labels.
 const (
-	// prometheus metrics
-	Mount                         = "mount"
-	RemoteRegistryGet             = "remote_registry_get"
-	NodeReaddir                   = "node_readdir"
-	InitMetadataStore             = "init_metadata_store"
-	ReadOnDemand                  = "read_on_demand"
-	MountLayerToLastOnDemandFetch = "mount_layer_to_last_on_demand_fetch"
+	/**
+	Metrics emitted via prometheus endpoint. All latency metrics are measured in milliseconds,
+	unless specified otherwise.
+	*/
+	// Mount measures the duration of Mount operation for resolved layer.
+	Mount = "mount"
+
+	// RemoteRegistryGet records the roundtrip latency for remote registry GET operation .
+	RemoteRegistryGet = "remote_registry_get"
+
+	// NodeReaddir measures how long readdir operation takes (in microseconds).
+	NodeReaddir = "node_readdir"
+
+	// InitMetadataStore measures how long it takes to initialize the file metadata.
+	InitMetadataStore = "init_metadata_store"
+
+	// ReadOnDemand measures the latency of on-demand reads (in mocroseconds).
+	ReadOnDemand = "read_on_demand"
 
-	OnDemandReadAccessCount          = "on_demand_read_access_count"
+	// OnDemandReadAccessCount records the count of on-demand reads.
+	OnDemandReadAccessCount = "on_demand_read_access_count"
+
+	// OnDemandRemoteRegistryFetchCount records the count of the on-demand fetch requests to remote registry.
 	OnDemandRemoteRegistryFetchCount = "on_demand_remote_registry_fetch_count"
-	OnDemandBytesServed              = "on_demand_bytes_served"
-	OnDemandBytesFetched             = "on_demand_bytes_fetched"
 
-	// logs metrics
-	BackgroundFetchTotal      = "background_fetch_total"
-	BackgroundFetchDownload   = "background_fetch_download"
+	// OnDemandBytesServed counts the number of bytes served on-demand.
+	OnDemandBytesServed = "on_demand_bytes_served"
+
+	// OnDemandBytesFetched counts the number of bytes fetched on-demand.
+	OnDemandBytesFetched = "on_demand_bytes_fetched"
+
+	/**
+	Metrics logged in snapshotter's log.
+	All latencies here are measured in milliseconds, unless stated otherwise.
+	*/
+	// MountLayerToLastOnDemandFetch measures the latency of last on demand fetch,
+	// which must be invoked at the end of background fetch operation only.
+	// Since this is expected to happen only once per container launch, it is recorded as a log line,
+	// instead of being emitted as a metric.
```

Comment:
It's not emitted as metric, so why is it included in the list of Metrics here?

Reply:
Because it's written in the log. It doesn't make sense to emit a metric, for a one-time event, that is why it is written in the log.

Reply:
Everything, related to background fetch is written in the log, since it's just 1 value per layer.